scripts/tls/README_TLS_ciphersuites

```diff
@@ -0,0 +1,32 @@
+TLS ciphers extraction
+======================
+
+:Date: 2014-03-26
+:Version: 1
+:Author: Pierre Chifflier <chifflier@wzdftpd.net>
+
+
+These scripts are used to auto-generate the YAML configuration file for
+suricata, containing the list of known TLS ciphersuites and parameters.
+
+The first script (extract-iana-ciphers.py) fetch the list from the IANA website
+(http://www.iana.org/assignments/tls-parameters/tls-parameters.xml), and tries
+to convert it to a CSV file. The names and parameters of ciphersuites are
+normalized, to ensure some consistency in the resulting values.
+
+.. note:: The format of the XML file provided by IANA has a lot of small
+  inconsistencies, especially in the values separated by underscores in names.
+  For this reasons, the scripts have to apply many workarounds, and will
+  probably need to be updated for every new version of the XML file.
+  The mapping between openssl ciphersuites names, and IANA names, is also
+  defined in these scripts.
```

Comment:
Do you mean the C code ? I don't feel comfortable loading an XML file and manipulating fields in Suricata, so I preferred transforming the XML file to a valid JSON struct.
Pushing the fields "as-is" in the JSON file is not really satisfying too, you'd still need to manipulated them at runtime.

Reply:
How often would this need to be run? IIRC @jasonish also considers to make this data file part of the files suricata-update will fetch/manage

Reply:
Updates are pretty rare, it happens only when a new cipher is updated (for ex, last time was the addition of the CHACHA20_POLY1305 family). Maybe it will be updated for TLS 1.3 (not sure, since the cipher IDs are already known).